src/metabase/util/honeysql_extensions.clj

```diff
@@ -231,128 +105,258 @@
-(def ^{:arglists '([& exprs])} month   "SQL `month` function."   (partial hsql/call :month))
-(def ^{:arglists '([& exprs])} quarter "SQL `quarter` function." (partial hsql/call :quarter))
-(def ^{:arglists '([& exprs])} year    "SQL `year` function."    (partial hsql/call :year))
-(def ^{:arglists '([& exprs])} concat  "SQL `concat` function."  (partial hsql/call :concat))
-
-;; Etc (Dev Stuff)
-
-(extend-protocol pretty/PrettyPrintable
-  honeysql.types.SqlCall
-  (pretty [{fn-name :name, args :args, :as this}]
-    (with-meta (apply list `hsql/call fn-name args)
-      (meta this))))
-
-(defmethod print-method honeysql.types.SqlCall
-  [call writer]
-  (print-method (pretty/pretty call) writer))
-
-(defmethod pprint/simple-dispatch honeysql.types.SqlCall
-  [call]
-  (pprint/write-out (pretty/pretty call)))
-
-(defmethod hformat/format-clause :returning [[_ fields] _]
-  (->> (flatten fields)
-       (map hformat/to-sql)
-       (hformat/comma-join)
-       (str "RETURNING ")))
+(defn abs
+  "SQL `abs` function."
+  [& exprs]
+  (case *honey-sql-version*
+    1 (apply h1x/abs exprs)
+    2 (apply h2x/abs exprs)))
+
+(defn ceil
+  "SQL `ceil` function."
+  [& exprs]
+  (case *honey-sql-version*
+    1 (apply h1x/ceil exprs)
+    2 (apply h2x/ceil exprs)))
+
+(defn floor
+  "SQL `floor` function."
+  [& exprs]
+  (case *honey-sql-version*
+    1 (apply h1x/floor exprs)
+    2 (apply h2x/floor exprs)))
+
+(defn second
+  "SQL `second` function."
+  [& exprs]
+  (case *honey-sql-version*
+    1 (apply h1x/second exprs)
+    2 (apply h2x/second exprs)))
+
+(defn minute
+  "SQL `minute` function."
+  [& exprs]
+  (case *honey-sql-version*
+    1 (apply h1x/minute exprs)
+    2 (apply h2x/minute exprs)))
+
+(defn hour
+  "SQL `hour` function."
+  [& exprs]
+  (case *honey-sql-version*
+    1 (apply h1x/hour exprs)
+    2 (apply h2x/hour exprs)))
+
+(defn day
+  "SQL `day` function."
+  [& exprs]
+  (case *honey-sql-version*
+    1 (apply h1x/day exprs)
+    2 (apply h2x/day exprs)))
+
+(defn week
+  "SQL `week` function."
+  [& exprs]
+  (case *honey-sql-version*
+    1 (apply h1x/week exprs)
+    2 (apply h2x/week exprs)))
+
+(defn month
+  "SQL `month` function."
+  [& exprs]
+  (case *honey-sql-version*
+    1 (apply h1x/month exprs)
+    2 (apply h2x/month exprs)))
+
+(defn quarter
+  "SQL `quarter` function."
+  [& exprs]
+  (case *honey-sql-version*
+    1 (apply h1x/quarter exprs)
+    2 (apply h2x/quarter exprs)))
+
+(defn year
+  "SQL `year` function."
+  [& exprs]
+  (case *honey-sql-version*
+    1 (apply h1x/year exprs)
+    2 (apply h2x/year exprs)))
+
+(defn concat
+  "SQL `concat` function."
+  [& exprs]
+  (case *honey-sql-version*
+    1 (apply h1x/concat exprs)
+    2 (apply h2x/concat exprs)))
+
+(defn at-time-zone
+  "Return a Honey SQL `expr` at time `zone`."
+  [expr zone]
+  (case *honey-sql-version*
+    1 (h1x/->AtTimeZone expr zone)
+    2 (h2x/at-time-zone expr zone)))
```

Comment:
Surprised you didn't write a macro that did the `case` for you, but I guess doing it all explicitly makes it sparklingly clear.